imports/client/components/SplitPanePlus.tsx

```diff
@@ -111,168 +154,175 @@ class SplitPanePlus extends React.Component<SplitPanePlusProps, SplitPanePlusSta
-    } else if (size + halfResizerSize >= fullSize - autoCollapseSecondary) {
-      // Collapse Secondary
+    } else if (autoCollapseSecondary >= 0 && size >= fullSize - autoCollapseSecondary) {
       return this.props.primary === 'first' ? 2 : 1;
     }
     return 0;
   }
 
   splitPaneNode(): HTMLElement | null {
-    if (!this.ref.current) {
-      return null;
-    }
-    if (!(this.ref.current.firstChild instanceof HTMLElement)) {
+    if (!this.ref.current || !(this.ref.current.firstChild instanceof HTMLElement)) {
       return null;
     }
     return this.ref.current.firstChild;
   }
 
-  primaryPaneNode() {
+  primaryPaneNode(): HTMLElement | null {
     const root = this.splitPaneNode();
     const className = `Pane${this.props.primary === 'first' ? 1 : 2}`;
     return root ? _.find(root.childNodes, (n) => (
       n instanceof Element &&
       n.classList.contains(className)
-    )) as Element : null;
+    )) as HTMLElement : null;
   }
 
-  secondaryPaneNode() {
+  secondaryPaneNode(): HTMLElement | null {
     const root = this.splitPaneNode();
     const className = `Pane${this.props.primary === 'first' ? 2 : 1}`;
     return root ? _.find(root.childNodes, (n) => (
       n instanceof Element &&
       n.classList.contains(className)
-    )) as Element : null;
+    )) as HTMLElement : null;
   }
 
-  resizerNode() {
+  resizerNode(): HTMLElement | null {
     const root = this.splitPaneNode();
     return root ? _.find(root.childNodes, (n) => (
       n instanceof Element &&
       n.classList.contains('Resizer')
-    )) as Element : null;
+    )) as HTMLElement : null;
   }
 
-  measure(elem: Element) {
+  measure(elem: HTMLElement | null): number {
     if (!elem) {
       return NaN;
     }
     return this.props.split === 'vertical' ? elem.clientWidth : elem.clientHeight;
   }
 
   render() {
-    const paneProps: SplitPaneProps = _.extend({}, this.props);
-
-    const defaultPaneStyles = { overflow: 'auto' };
-    const maxRangeStyles = (this.props.split === 'vertical' ?
-      { maxWidth: '100%' } :
-      { maxHeight: '100%' }
+    const paneProps: { [key: string]: any; } = _.pick(
+      this.props,
+      'children',
+      'primary',
+      'split',
+      'allowResize',
+      'step',
+      'minSize',
+      'maxSize',
+      'className',
+      'style',
+      'paneClassName',
+      'paneStyle',
+      'pane1ClassName',
+      'pane1Style',
+      'pane2ClassName',
+      'pane2Style',
+      'resizerClassName',
+      'resizerStyle',
     );
-    paneProps.pane1Style = _.extend({},
-      defaultPaneStyles,
-      this.props.primary === 'first' ? maxRangeStyles : {},
-      this.props.pane1Style);
-    paneProps.pane2Style = _.extend({},
-      defaultPaneStyles,
-      this.props.primary === 'first' ? {} : maxRangeStyles,
-      this.props.pane2Style);
-
+    const defaultPaneStyle: React.CSSProperties = { overflow: 'auto' };
+    paneProps.paneStyle = _.extend(defaultPaneStyle, this.props.paneStyle);
```

Comment:
FWIW, I think this will mutate `defaultPaneStyle`, which is probably not what you want. `{ ...defaultPaneStyle, ...this.props.paneStyle }` should work to create a new object.

(I also kiiiind of want to move away from using underscore because the type definitions are a bit weaker than a lot of the modern ES6+ native equivalents - e.g. `_.extend` is defined as taking `any` and returning `any`, which isn't super helpful)

Reply:
Oh I...missed that you were just defining this above. Still could probably use object-spread syntax.

Reply:
Just sloppy on my part. Good catch. Here it was functionally fine, but a few lines below pane1Style could leak into pane2Style.

I had no idea you could do that with object spread. Neat.